osu.Game/Online/Multiplayer/MultiplayerRoom.cs

```diff
@@ -13,35 +14,42 @@ namespace osu.Game.Online.Multiplayer
     /// A multiplayer room.
     /// </summary>
     [Serializable]
+    [MessagePackObject]
     public class MultiplayerRoom
     {
         /// <summary>
         /// The ID of the room, used for database persistence.
         /// </summary>
+        [Key(0)]
         public readonly long RoomID;
 
         /// <summary>
         /// The current state of the room (ie. whether it is in progress or otherwise).
         /// </summary>
+        [Key(1)]
         public MultiplayerRoomState State { get; set; }
 
         /// <summary>
         /// All currently enforced game settings for this room.
         /// </summary>
+        [Key(2)]
         public MultiplayerRoomSettings Settings { get; set; } = new MultiplayerRoomSettings();
 
         /// <summary>
         /// All users currently in this room.
         /// </summary>
+        [Key(3)]
         public List<MultiplayerRoomUser> Users { get; set; } = new List<MultiplayerRoomUser>();
 
         /// <summary>
         /// The host of this room, in control of changing room settings.
         /// </summary>
+        [Key(4)]
         public MultiplayerRoomUser? Host { get; set; }
 
         [JsonConstructor]
-        public MultiplayerRoom(in long roomId)
+        [SerializationConstructor]
+        public MultiplayerRoom(long roomId)
```

Comment:
More out of curiosity than anything else, but is there a reason for dropping this `in`? Is that just not supported by messagepack or something?

Reply:
Yeah, looks to be an oversight at their end. Maybe worth opening an issue for, maybe not.